lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -365,26 +408,54 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
 
     /// Create assignments to the inputs of the library op.
     auto group = createGroupForOp<TGroupOp>(rewriter, op);
+
+    if (isSequential) {
+      auto groupOp = cast<calyx::GroupOp>(group);
+      getState<ComponentLoweringState>().addBlockScheduleable(op->getBlock(),
+                                                              groupOp);
+    }
+
     rewriter.setInsertionPointToEnd(group.getBodyBlock());
-    for (auto dstOp : enumerate(opInputPorts))
-      rewriter.create<calyx::AssignOp>(op.getLoc(), dstOp.value(),
-                                       op->getOperand(dstOp.index()));
+
+    for (auto dstOp : enumerate(opInputPorts)) {
+      if (isSequential)
```

Comment:
Bug, should be:
```suggestion
      if (isPipeLibOp(dstOp.value()))
        rewriter.create<calyx::AssignOp>(op.getLoc(), dstOp.value(),
                                         srcReg.getOut());
```

Then seems like we are checking `isPipeLibOp` twice, once in a caller function, another in a callee..